packages/client/test/unit/util/typeCasting.test.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  Bool,
+  Field,
+  Float64,
+  Int64,
+  Timestamp,
+  TimeUnit,
+  Uint64,
+  Utf8,
+} from 'apache-arrow'
+import {getMappedValue} from '../../../src/util/TypeCasting'
+import {expect} from 'chai'
+
+describe('Type casting test', () => {
```

Comment:
These tests might also be improved using dynamic test generation and a `forEach` loop.